John/mention integration

jt5301 opened the pull request:
Sorting functions for the most popular / most recent toggle, integrated mentions frontend with routes
<img width="1530" alt="Screen Shot 2020-07-09 at 5 02 36 PM" src="https://user-images.githubusercontent.com/47200207/87090487-12d89e80-c206-11ea-9a69-399e817663dc.png">

jt5301 committed:
sortby and mentnion integration

jt5301 committed:
integration, sort toggle

jt5301 committed:
sorted + backend integration

jt5301 committed:
minor fixes

jt5301 committed:
sorting functions for toggle, integrated mentions for server data

jt5301 committed:
added uuid for mapping

powerseed commented on client/src/App.js:
```diff
@@ -5,7 +5,7 @@ import { theme } from "./themes/theme";
 import Login from './pages/Login'
 import Main from './pages/Main'
-import Header from "./components/Header"
+import Header from "./Components/Header"
```
I think we would better use lowercase.
Sorry it is me who started it...

jt5301 replied:
done!

powerseed commented on client/src/Components/Header.js:
```diff
@@ -22,7 +22,7 @@ const useStyles = makeStyles(() => ({
     height: "5.4em",
     boxShadow: "none",
-    position: "fixed",
+    // position: "fixed",
```
May I know the reason why change it to be not fixed? Have you tried it in main page with more than one page of mentions? I guess the header will move when you scroll down?

jt5301 replied:
The toggle doesn't show up for some reason if the position is fixed, but I assume we want the header to be stuck on top? Right now the header will not follow if there are multiple mentions. I think this will be a temporary fix for the demonstration to show the toggle, but I can try to have it fixed without blocking any other components moving forward

powerseed replied:
Sry I stated it wrongly. We may NOT want the header to be stuck on top, otherwise it would be like this:
![image](https://user-images.githubusercontent.com/42278237/87100958-4d950380-c213-11ea-8c86-fb9d4e3ab73f.png)
There is an empty strip at the top of the screen, with the left panel displayed it is strange.
The 2nd reason is that the search bar is in the header, and we want the user to have access to the search bar no matter how much she has scroll down. Otherwise she would have to scroll all the way up to the top to enter some new keywords.

jt5301 replied:
my bad, yes that's what I meant when I said stuck, that it is always reachable on top by the user. Let me try playing with css before merging.

powerseed commented on client/src/pages/Login.js:
```diff
@@ -2,8 +2,8 @@ import React, { useState, } from 'react';
 
-import SubmitButton from '../components/SubmitButton'
-import CustomTextField from '../components/CustomTextField'
+import SubmitButton from '../Components/SubmitButton'
+import CustomTextField from '../Components/CustomTextField'
```
same as previous comment

jt5301 replied:
done!

powerseed commented on client/src/pages/Main.js:
```diff
@@ -1,8 +1,7 @@
-import Mention from "../components/Mention";
-import Grid from "@material-ui/core/Grid";
-
+import Mention from "../Components/Mention";
```
Same here

jt5301 replied:
done!

powerseed commented on client/src/pages/Main.js:
```diff
@@ -122,107 +122,37 @@ class Main extends Component {
+
+  sortByDate(mentions) {
+    mentions.sort((a, b) => {
+      return new Date(b.date) - new Date(a.date)
```
The `date` field of mentions is in unix format, aka, the millisecs that have pasted from 1970 to now, so I think you can just compare them directly without using `new Date()`.
I didn't try it, correct me if I am wrong

jt5301 replied:
Ahh I see. I used postico to hardcode the date which isn't unix format. I will change this

powerseed commented on client/src/pages/Signup.js:
```diff
@@ -2,8 +2,8 @@ import React, { useState, } from 'react';
-import SubmitButton from '../components/SubmitButton'
-import CustomTextField from '../components/CustomTextField'
+import SubmitButton from '../Components/SubmitButton'
+import CustomTextField from '../Components/CustomTextField'
```
Please change them to lowercase. I am sorry again...

jt5301 replied:
done!

jt5301 committed:
added padding on top

jt5301 commented:
@powerseed ok, added some padding some that the header can keep position:fixed and the right panel is unobstructed:
<img width="1680" alt="Screen Shot 2020-07-09 at 9 08 10 PM" src="https://user-images.githubusercontent.com/47200207/87105311-94412880-c228-11ea-867e-4c47cc7d7188.png">

Nice. Please merge if all are resolved.

jt5301 committed:
Merge branch 'dev' into john/mention_integration
